pkg/skaffold/runner/runner.go

```diff
@@ -128,25 +132,97 @@ func NewForConfig(opts *config.SkaffoldOptions, cfg *latest.SkaffoldConfig) (*Sk
 
 	monitor := filemon.NewMonitor()
 
-	return &SkaffoldRunner{
+	intentChan := make(chan bool, 1)
```

Comment:
can this channel live inside `intents`?

Reply:
no, it needs to live inside the `Listener` because that's where it's consumed, and the listener doesn't have access to the `Runner`. I made the channel read-only inside the listener to make its purpose more clear.